tests/integration/rules_test.go

```diff
@@ -0,0 +1,467 @@
+package integration_test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	json "github.com/bitly/go-simplejson"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	oathkeeperv1alpha1 "github.com/ory/oathkeeper-k8s-controller/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	namespaceName                 = "test-namespace" //TODO: Randomize?
+	defaultTargetMapNamespace     = "oathkeeper-k8s-controller-system"
+	defaultTargetMapName          = "oathkeeper-rules"
+	maxRetriesWaitingForConfigMap = 35 //Twice as max registered on my machine.
+	maxRetriesWaitingForRule      = 10
+)
+
+var (
+	k8sClient        *kubernetes.Clientset       = getK8sClientOrDie()
+	k8sDynamicClient dynamic.Interface           = getK8sDynamicClientOrDie()
+	ruleResource     schema.GroupVersionResource = schema.GroupVersionResource{Group: oathkeeperv1alpha1.GroupVersion.Group, Version: oathkeeperv1alpha1.GroupVersion.Version, Resource: "rules"}
+)
+
+var _ = BeforeSuite(func() {
+	//Create namespace
+	_, err := createNamespace(namespaceName, k8sClient)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	//Delete namespace
+	err := deleteNamespace(namespaceName, k8sClient)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("Oathkeeper controller", func() {
+
+	Context("in happy path scenario", func() {
+
+		It("should", func() {
```

Comment:
how 'bout:
```
var _ = Describe("Oathkeeper controller", func() {

	It("should represent all in-cluster oathkeeper rules as a single entry in a Configmap", func() {

		By("creating a valid Configmap upon first reconciliation")
```

etc.

Reply:
After thinking about it, I chose this version:
```
Oathkeeper controller
  should manage rules as ConfigMap entries
    in happy path scenario
```